assets/src/components/query/limit/index.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * USAGE:
+ *
+ * A React component for adding a WordPress Range Control input
+ * for setting a query's limit clause
+ *
+ * Step 1: import the following external dependencies:
+ *
+ *        import { __ } from '@eventespresso/i18n';
+ *        import { QueryLimit } from '@eventespresso/components';
+ *        import { PanelBody } from '@wordpress/components';
+ *        import { InspectorControls } from '@wordpress/editor';
+ *
+ * Step 2: add a "limit" attribute:
+ *
+ *        attributes: {
+ * 		  	limit: {
+ * 		  		type: 'number',
+ * 		  		default: 5, (10 is already set as default)
+ * 		  	},
+ * 		},
+ *
+ * Step 3: wrap the QueryLimit component within
+ * the InspectorControls and PanelBody components
+ * (currently shown within a block's edit() function)
+ *
+ *        edit: ( ( { attributes, setAttributes } ) => {
+ * 			const { limit } = attributes;
+ * 			return (
+ * 				<InspectorControls>
+ * 		  			<PanelBody>
+ * 		  				<QueryLimit
+ * 		  					limit={ limit }
+ * 		  					onLimitChange={
+ * 		  						( value ) => setAttributes( { limit: value } )
+ * 		  					}
+ * 		  				/>
+ * 		  			</PanelBody>
+ * 		  		</InspectorControls>
+ * 		  	);
+ * 		}),
+ */
```

Comment:
I prefer not to have documentation like this in the component.  Having the documentation in our docs folder (as you did) is sufficient or we could adopt the same pattern as GB where there's a README.md located in the same directory that has documentation for the things in that directory.  That might work better for js documentation as we remove the entire source directory for release builds.

Reply:
Is there an actual reason other than preference?

I added this here so that someone looking through our components folder that wanted to use `QueryLimit` would know how to do so without having to go searching elsewhere for documentation about it. So maybe having a README.md here would be good as well as the other documentation I provided (ie: just duplicate all block/component documentation so that there are local in place READMEs as well as the docs folder where everything is combined).

We may even want to follow this pattern for PHP and include READMEs within individual folders, so for example the `core/services/commands` folder could have it's own README.md which would be a duplicate (or link to) the existing documenation at `\docs\N--Core-Functionality\command-bus-hexagonal-architecture.md`.

thoughts?